internal/controller/awsvalidator_controller.go

```diff
@@ -115,22 +110,73 @@ func (r *AwsValidatorReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	return ctrl.Result{RequeueAfter: time.Second * 120}, nil
 }
 
-// envFromSecret sets environment variables from a secret to configure AWS credentials
-func (r *AwsValidatorReconciler) envFromSecret(name, namespace string) error {
-	r.Log.Info("Configuring environment from secret", "name", name, "namespace", namespace)
+// configureAwsAuth sets environment variables to control AWS authentication. Order of precedence
+// for source:
+// 1 - Kubernetes secret
+// 2 - Specified inline in spec
+// Returns an error if env vars couldn't be set for any reason.
+func (r *AwsValidatorReconciler) configureAwsAuth(auth v1alpha1.AwsAuth, reqNamespace string, l logr.Logger) error {
+	if auth.Implicit {
+		l.Info("auth.implicit set to true. Skipping setting AWS env vars.")
+		return nil
+	}
+
+	if auth.Credentials == nil {
+		auth.Credentials = &v1alpha1.Credentials{}
+	}
+
+	// If Secret name provided, override any env var values with values from its data.
+	if auth.SecretName != "" {
+		l.Info("auth.secretName provided. Using Secret as source for any AWS env vars defined in its data.", "secretName", auth.SecretName, "secretNamespace", reqNamespace)
+		nn := ktypes.NamespacedName{Name: auth.SecretName, Namespace: reqNamespace}
+		secret := &corev1.Secret{}
+		if err := r.Get(context.Background(), nn, secret); err != nil {
+			return fmt.Errorf("failed to get Secret: %w", err)
+		}
+		if accessKeyID, ok := secret.Data["AWS_ACCESS_KEY_ID"]; ok {
+			l.Info("Using access key ID from Secret.")
+			auth.Credentials.AccessKeyID = string(accessKeyID)
+		}
+		if secretAccessKey, ok := secret.Data["AWS_SECRET_ACCESS_KEY"]; ok {
+			l.Info("Using secret access key from Secret.")
+			auth.Credentials.SecretAccessKey = string(secretAccessKey)
+		}
```

Comment:
We may as well support grabbing the session token from the secret, right?

Reply:
Yes! Thanks for catching that. Added that now. Just waiting for CI/CD again.